### PR TITLE
improve docs to point to memory leak if ngOnDestroy is overridden in …

### LINKED ISSUE #3982: Potential Memory Leak Issue in ngrx Store-Component
### Information

I am writing to bring to your attention a potential memory leak issue that I discovered in the ngrx store component. 
As an avid user of the library, I noticed some behaviors that suggest there might be a problem when overriding the ngOnDestroy method, 
where the parent class does not release resources properly.

Description of the Issue:
Upon thorough analysis, I suspect that the store component may be causing a memory leak when the ngOnDestroy method is overridden. The parent class will not be called, and as a result and resources will not release, which may lead to inefficient memory utilization.

Expected Behavior:
When overriding the onDestory you should add super.onDestory()

Relevant Code Snippet:

` /** Completes all relevant Observable streams. */
  ngOnDestroy() {
    this.stateSubject$.complete();
    this.destroySubject$.next();
  }`
  
Impact:
A memory leak issue can severely impact the performance and stability of applications using the ngrx store-component. It can lead to increased memory consumption, decreased performance, and even application crashes in severe cases.

### Documentation page

https://v15.ngrx.io/guide/component-store/lifecycle#ondestroy

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [X] No

### PR DESCRIPTION
**Pull request by ValentinBossi**
…own component store

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #3982

## Does this PR introduce a breaking change?

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/component-store/lifecycle.md [hunk 1]
```diff
@@ -14,6 +14,18 @@ Currently, Angular provides initializer tokens in a few areas. The `APP_INITIALI
 
 </div>
 
+<div class="alert is-important">
+
+**Note:** If you override the `ngOnDestroy` method in your component store, you need to call `super.ngOnDestroy()`. Otherwise a memory leak may occur.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/component-store/lifecycle.md:19]: This would go better down at the bottom with the `OnDestroy` section.

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: projects/ngrx.io/content/guide/component-store/lifecycle.md (hunk 1 above)
Comment (ValentinBossi) [reply projects/ngrx.io/content/guide/component-store/lifecycle.md]: absolutely. i changed that.

File: projects/ngrx.io/content/guide/component-store/lifecycle.md [hunk 2]
```diff
@@ -138,6 +138,18 @@ ComponentStore also implements the `OnDestroy` interface from `@angulare/core` t
 
 It also exposes a `destroy$` property on the ComponentStore class that can be used instead of manually creating a `Subject` to unsubscribe from any observables created in the component.
 
+<div class="alert is-important">
```
[the remark this answers is not in the document]
Comment (ValentinBossi) [reply projects/ngrx.io/content/guide/component-store/lifecycle.md:141]: ok i added the code snipped

**Review by timdeschryver — APPROVED** [review]
Thanks @ValentinBossi !

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**